static/js/generic_checkbox.html

```diff
@@ -0,0 +1,309 @@
+			vars[key] = value;
+		  });
+		  return vars;
+		}
+
+		/*
+			This is a global state dictionary. We define this for convenience when working
+			with edX's state-saving paradigm. Any data that is required for operation to
+			seamlessly persist across page refreshes, or that EdX needs to know about, must
+			be placed in here.
+		*/
+		state = {
+			//is the student's answer correct? (possibly a misnomer... eh)
+			//required for grading the problem
+			'correct_answer' : false,
+			//which choice is the answer?
+			'answer' : -1,
+			//list of user's answer choices
+			//required to repopulate the form so users don't shit bricks
+			'inputs' : [],
+			//attempts at the problem (how many times getGrade() has been called)
+			//required for our adaptive hints
+			'attempts' : 0,
+			//number of choices
+			'choices' : 0
+		};
+
+		//global variable that is set once the form is displayed.
+		//If this is false we can't access the form's document elements,
+		//because they haven't yet been created.
+		displayed = false;
+
+		//url variables
+		vars = {};
+
+		function getVars() {
+			vars = getUrlVars();
+		}
+
+		//Compose answer strings for form creation
+		function composeAnswerStringList(div) {
+			var answerlist = [];
+			var row;
+			for (row = 0; row < state['choices']; row++) {
+				var formstring = '';
+				formstring += '<input type="checkbox" name = "ans" id="ans'+row.toString();
+				formstring += '" onkeypress= "setTimeout(check_answers, 250)"/>';
+				if(div.hasAttribute('choice'+row.toString())) {
+					formstring += div.getAttribute('choice'+row.toString()) + '<br>';
+				}
+				else {
+					formstring += '<i>no answer</i><br/>';
+				 }
+				answerlist.push(formstring);
+			}
+			return answerlist;
+		}
+
+		//Shuffle answerlist items in composed form for answers
+		function shuffleString(iterObject) {
+			var steps;
+			var temp = iterObject.slice(0);
+			var cacheString = '';
+			for (steps = iterObject.length; steps > 0; steps--) {
+				var rand = Math.floor(Math.random() * steps);
+				cacheString += temp.splice(rand, 1);
+			}
+			return cacheString;
+		}
+
+		//Creates the form where users input data. Super simple
+		function createForm(divname) {
+			var div = parent.document.getElementById(vars['div']);
+			state['choices'] = parseInt(div.getAttribute('choices'));
+			var prompt = '';
+			console.log("Creating checkbox form");
+			if(div.hasAttribute('text'))
+			{
+				prompt = div.getAttribute('text') + '<br>';
+			}
+			var formstring = prompt + '<form>';
+			var answerlist = composeAnswerStringList(div);
+
+			formstring += shuffleString(answerlist);
+
+			formstring += '</form>';
+
+			var form = document.getElementById(divname);
+			form.innerHTML = formstring;
+		}
+
+		//create list with correct answers
+		function correctAnswerList(){
+			var anslist = [];
+			var row;
+			for (row = 0; row < state['choices']; row++){
+				if (state['answer'].indexOf(row) > -1) {
+					anslist.push(1)
```

Comment:
1 and 0 are numeric representation of True and False correspondingly and used widely.